src/mappings/rewardsDistributor.ts

```diff
@@ -90,7 +94,33 @@ export async function handleRewardsDistributed(
       );
       await Delegation.remove(delegator.id);
     }
+
+    if (rewards.gt(0)) {
+      const eraReward = await updateEraReward({
+        indexerId: indexer,
+        delegatorId: delegator.delegatorId,
+        eraId: eraIdx.toHexString(),
+        isCommission: false,
+        claimed: false,
+        amount: rewards.toBigInt() - rewardOld,
```

Comment:
The reward from this event is the reward of this era, so you don't need to minus it by rewardOld
```suggestion
        amount: rewards.toBigInt(),
```